prost-derive/src/lib.rs

```diff
@@ -91,16 +90,19 @@ fn try_message(input: TokenStream) -> Result<TokenStream, Error> {
     fields.sort_by_key(|(_, field)| field.tags().into_iter().min().unwrap());
     let fields = fields;
 
-    let mut tags = fields
+    if let Some((duplicate_tag, _)) = fields
         .iter()
         .flat_map(|(_, field)| field.tags())
-        .collect::<Vec<_>>();
-    let num_tags = tags.len();
-    tags.sort_unstable();
-    tags.dedup();
-    if tags.len() != num_tags {
-        bail!("message {} has fields with duplicate tags", ident);
-    }
+        .sorted_unstable()
+        .tuple_windows()
+        .find(|(a, b)| a == b)
+    {
+        bail!(
+            "message {} has multiple fields with tag {}",
```

Comment:
So this basically improves the error message. Do I understand correctly?

Reply:
correct :+1:  and i feel the expression of the inadmissible condition is a little more straightforward. "deduplicated list is the same size as the original list" is perfectly valid, but it feels unnecessarily roundabout when we can find duplicates directly.